src/main/java/com/hubspot/ringleader/watcher/PersistentWatcher.java

```diff
@@ -79,15 +79,15 @@ public void start() {
     if (started.compareAndSet(false, true)) {
       fetchInExecutor();
 
-      executor.schedule(() -> {
+      executor.scheduleAtFixedRate(() -> {
```

Comment:
Can anything in here throw an exception and cancel the scheduled executions?

Reply:
oh yeah definitely, thanks for catching that